### DISCUSSION TITLE
Calling a scenario within a step

### DISCUSSION THREAD

**Discussion by ARK215** [Tyler-Keith-Thompson/CucumberSwift#48] [Q&A] [answered]
Hi!
I have a login screen, which is shown on every app launch. I implemented passing this screen as a scenario, but I need to call those actions again for every further test.
Is there a possibility to call the already implemented scenario within a step?

**Comment by Tyler-Keith-Thompson** [comment, marked as answer]
So I can think of a few ways to do this cleanly. One is using [Background](https://cucumber.io/docs/gherkin/reference/#background) with your Gherkin, it allows you to define a set of steps that should happen before every scenario.

You could also extract your steps to functions, and then call those functions as needed.

Finally, you can use [hooks](https://tyler-keith-thompson.github.io/CucumberSwift/documentation/cucumberswift/hooks) as a way to execute logic before a scenario runs.

All of these should give you the behavior you desire.
